bundles/mapping/layerswipe/instance.js

```diff
@@ -0,0 +1,143 @@
+import { getRenderPixel } from 'ol/render';
+import { unByKey } from 'ol/Observable';
+import VectorLayer from 'ol/layer/Vector';
+
+Oskari.clazz.define(
+    'Oskari.mapframework.bundle.layerswipe.LayerSwipeBundleInstance',
+
+    function () {
+        this.active = false;
+        this.splitter = null;
+        this.splitterWidth = 5;
+        this.cropSize = null;
+        this.map = null;
+        this.loc = Oskari.getMsg.bind(null, 'LayerSwipe');
+        this.eventListenerKeys = [];
+    },
+    {
+        __name: 'LayerSwipe',
+
+        _startImpl: function (sandbox) {
+            this.map = Oskari.getSandbox().findRegisteredModuleInstance('MainMapModule').getMap();
+
+            const addToolButtonBuilder = Oskari.requestBuilder('Toolbar.AddToolButtonRequest');
+            const buttonConf = {
+                iconCls: 'tool-layer-swipe',
+                tooltip: this.loc('toolLayerSwipe'),
+                sticky: true,
+                callback: () => this.setActive(!this.active)
+            };
+            sandbox.request(this, addToolButtonBuilder('LayerSwipe', 'basictools', buttonConf));
+        },
+
+        setActive: function (active) {
+            if (active) {
+                const layer = this.getTopmostLayer();
+                if (layer) {
+                    this.registerEventListeners(layer);
+                }
+                this.showSplitter();
+                if (this.cropSize === null) {
+                    const mapSize = this.map.getSize();
+                    this.cropSize = mapSize[0] / 2;
+                }
+            } else {
+                this.unregisterEventListeners();
+                this.hideSplitter();
+            }
+            this.active = active;
+            this.map.render();
+        },
+
+        updateSwipeLayer: function () {
+            this.unregisterEventListeners();
+            const layer = this.getTopmostLayer();
+            if (layer) {
+                this.registerEventListeners(layer);
+            }
+        },
+
+        getTopmostLayer: function () {
+            const layers = this.map.getLayers().getArray().filter(layer => layer.getVisible() && !(layer instanceof VectorLayer));
```

Comment:
Should the localized tooltip say that vectorlayers are not counted as layers? Or why do we filter them out? There might be some layers that we do want to filter out like a draw layer or marker layer but probably not all VectorLayers?